src/org/vitrivr/cineast/core/config/DatabaseConfig.java

```diff
@@ -0,0 +1,67 @@
+package org.vitrivr.cineast.core.config;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+public class DatabaseConfig {
+
+  private final String host;
+  private final int port;
+  private final boolean plaintext;
+  private final int batchsize;
+  private final Writer writer;
+  private final Selector selector;
+
+  @JsonCreator
+  public DatabaseConfig(
+      @JsonProperty("host") String host,
+      @JsonProperty("port") int port,
+      @JsonProperty("plaintext") boolean plaintext,
+      @JsonProperty("batchsize") int batchsize,
+      @JsonProperty("writer") Writer writer,
+      @JsonProperty("selector") Selector selector) {
+    this.host = host;
+    this.port = port;
+    this.plaintext = plaintext;
+    this.batchsize = batchsize;
+    this.writer = writer;
+    this.selector = selector;
+  }
+
+  public String getHost() {
+    return host;
+  }
+
+  public int getPort() {
+    return port;
+  }
+
+  public boolean isPlaintext() {
+    return plaintext;
+  }
+
+  public int getBatchsize() {
+    return batchsize;
+  }
+
+  public Writer getWriter() {
+    return writer;
+  }
+
+  public Selector getSelector() {
+    return selector;
+  }
+
+  public enum Writer {
+    PROTO,
+    JSON,
+    ADAMPRO
+  }
+
+  public enum Selector {
+    NONE,
+    JSON,
+    PROTO,
+    ADAMPRO
```

Comment:
Maybe have a look at the cineast 'dev' branch, there are more options there already.

Reply:
Will update this config to match the dev branch